azure-kusto-ingest/tests/test_kusto_ingest_client.py

```diff
@@ -156,6 +191,36 @@ def test_sanity_ingest_from_file(self, mock_uuid, mock_put_message_in_queue, moc
 
         assert type(upload_blob_kwargs["data"]) == io.BytesIO
 
+    @responses.activate
+    def test_ingest_from_file_wrong_endpoint(self):
+        responses.add_callback(
+            responses.POST, "https://somecluster.kusto.windows.net/v1/rest/mgmt", callback=request_error_callback, content_type="application/json"
+        )
+
+        ingest_client = QueuedIngestClient("https://somecluster.kusto.windows.net")
+        ingestion_properties = IngestionProperties(database="database", table="table", data_format=DataFormat.CSV)
+
+        current_dir = os.getcwd()
+        path_parts = ["azure-kusto-ingest", "tests", "input", "dataset.csv"]
+        missing_path_parts = []
+        for path_part in path_parts:
+            if path_part not in current_dir:
+                missing_path_parts.append(path_part)
+
+        file_path = os.path.join(current_dir, *missing_path_parts)
+
+        reached_exception = False
+        try:
+            ingest_client.ingest_from_file(file_path, ingestion_properties=ingestion_properties)
+        except Exception as e:
+            reached_exception = True
+            assert e.__class__ == KustoClientError
+            assert (
+                e.args[0]
+                == "You are using 'DataManagement' client type, but the provided endpoint is of ServiceType 'Engine'. Initialize the client with the appropriate endpoint URI: 'https://ingest-somecluster.kusto.windows.net'"
+            )
```

Comment:
newline after scope exit

Reply:
I don't know the Python conventions. Is there a specific guide/lint we're using? Because the lint I use says specifically to use 1 line there.

Here's the video capture: https://microsoft-my.sharepoint.com/:v:/p/yischoen/EVHw4mswbJZPppTJUhlcrZYBMRcfpIsltM9JyfijsldcEQ?e=MAZWt0

Reply:
We lint with black
pip install black

Reply:
Weird, I used black on that file, and this was the result. Also, it seems GitHub blocks merging if it fails black, right?

Reply:
Unless you say otherwise, I'm leaving this as I wrote it because I think I did it as black requires.